wagtail/admin/templates/wagtailadmin/reports/site_history.html

```diff
@@ -24,14 +24,20 @@
                 </tr>
             </thead>
             <tbody>
-                {% for entry in object_list %}
+                {% for entry, instance in object_list %}
```

Comment:
Sorry, very much not keen on this...

We went to a lot of trouble in 2.15 to remove the special-casing of the Page model in this report, and ensure that the 'log entry' records we pass to this template provide a consistent set of data regardless of the object instances they represent. Having the template do this kind of ad-hoc digging into instance properties goes against this goal.

I'd question whether having different object types having different HTML representations in this report is a good idea - it feels to me like it goes against the idea of a report providing consistent structured data. If we are going down that path, it probably calls for a general-purpose mechanism for doing that - perhaps a method on LogEntry that returns a [component object](https://docs.wagtail.io/en/stable/extending/template_components.html). Otherwise, I think we could consider a more 'lo-fi' way of including the locale information in the existing structure. Perhaps instead of outputting the `entry.label` field directly, LogEntry could provide a `display_label` property that defaults to returning `entry.label`, but can be overridden so that PageLogEntry returns title plus locale? (But then you'd have the problem of where to cache the locale ID to name lookup table, I guess.)

Reply:
I went round and round on this.

It is worth highlighting that this approach works for non-Page models that use [`TranslateableMixin`](https://github.com/wagtail/wagtail/blob/main/wagtail/core/models/i18n.py#L87)

The ideal place would be `LogEntry`, but we do have the problem of the cache as you highlight this. Given `TranslateableMixin` can be added to most other non-Page models, we'd need to cover this in `ModelLogEntryManager` as well.

Will have another look to see if we can arrive at something decent

Reply:
Been tinkering with this more and imho this is still the simpler solution that does not add loads of overhead. For example, `ModelLogEntry` only has the content type reference and the object id. To get an appropriate display label, we'd need load the instance in order to get locale id. Which means we hit a nice n+1 issue.

I looked at some of my earlier notes and adding a `select_related` in the queryset only improves things a small amount, but still constitutes a significant increase.

Doing it the current way, ie. passing the log item + source instance in the object list solves the n+1 issue, and is not a massive overhead since we process those items anyway in https://github.com/wagtail/wagtail/blob/main/wagtail/admin/views/reports/audit_logging.py#L168-L178

To properly change the approach, we'd want to capture the locale id at `log_action` time alongside the object/page id, if in use. This means 2 migrations (one to add the field, and the other to populate the locale value for existing log entries) 🤔

Reply:
hmm, I missed the `Component` suggestion. Would creating a `LocaleLabelComponent` that takes takes `instance` in context and takes care of the label rendering work?

We have this pattern in the other reports though, so may be one to tackle in the report componentization work?

I could move more of this logic to a template tag (again getting the instance as we do now) so it doesn't happen so much in the template...

Reply:
dropped the changes for site history